### PR TITLE
Feature: GET_OLD_BLOCK_NUMBER_AND_HASH hint

### PR DESCRIPTION
**Pull request by odesenfans**
Issue Number: N/A

## Type

- [x] feature
- [ ] bugfix
- [ ] dev (no functional changes, no API changes)
- [ ] fmt (formatting, renaming)
- [ ] build
- [ ] docs
- [ ] testing

## Breaking changes?

- [x] yes (one field added to ExecutionHelper)
- [ ] no

### CODE REVIEW

File: src/execution/helper.rs [hunk 1]
```diff
@@ -77,6 +83,14 @@ impl ExecutionHelperWrapper {
             })),
         }
     }
+
+    pub fn get_old_block_number_and_hash(&self) -> Result<(Felt252, Felt252), HintError> {
+        let eh_ref = self.execution_helper.as_ref().borrow();
+        eh_ref.old_block_number_and_hash.ok_or(HintError::AssertionFailed(
+            format!("Block number is probably < {STORED_BLOCK_HASH_BUFFER}.").into_boxed_str(),
```
Comment (maciejka) [inline src/execution/helper.rs:90]: Does this error message make sense?

File: src/execution/helper.rs [hunk 2]
```diff
@@ -34,6 +35,10 @@ pub struct ExecutionHelper {
     // Pointer to the Cairo ExecutionInfo struct of the current call.
     // Must match the ExecutionInfo pointer for system call validation in 'enter_call'
     pub call_execution_info_ptr: Option<Relocatable>,
+    // The block number and block hash of the (current_block_number - buffer) block, where
+    // buffer=STORED_BLOCK_HASH_BUFFER.
+    // It is the hash that is going to be written by this OS run.
+    pub old_block_number_and_hash: Option<(Felt252, Felt252)>,
```
Comment (maciejka) [inline src/execution/helper.rs:41]: When is it supposed to be `None`?

File: src/execution/helper.rs (hunk 1 above)
Comment (notlesh) [reply src/execution/helper.rs:90]: The original hint has a pretty specific error message: `"Inconsistent block number. The constant STORED_BLOCK_HASH_BUFFER is probably out of sync."`

Comment (maciejka) [reply src/execution/helper.rs:90]: Here we just unwrap old_block_number_and_hash. If it is None is it an internal error, which has nothing to do with block number being < stored block number.

File: src/execution/helper.rs (hunk 2 above)
Comment (odesenfans) [reply src/execution/helper.rs:41]: That's a good question, it's just optional in the Python VM so I kept the implementation. Seems weird to me as well.

File: src/execution/helper.rs (hunk 1 above)
Comment (odesenfans) [reply src/execution/helper.rs:90]: Yeah seems like I skipped a line here. Will fix it.

Comment (odesenfans) [reply src/execution/helper.rs:90]: Nevermind, this is translated directly from `cairo-lang`: 

https://github.com/starkware-libs/cairo-lang/blob/ee7ce74e1159a349d4b77a5f952241b50b1692de/src/starkware/starknet/core/os/syscall_handler.py#L1237

**Review by maciejka — APPROVED** [review]

**Pull request merged**